spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestSetWriteDistributionAndOrdering.java

```diff
@@ -249,6 +267,13 @@ public void testSetWriteDistributedByWithLocalSort() {
 
     SortOrder expected = SortOrder.builderFor(table.schema()).withOrderId(1).asc("id").build();
     assertThat(table.sortOrder()).as("Sort order must match").isEqualTo(expected);
+
+    sql("ALTER TABLE %s WRITE LOCALLY ORDERED BY id", tableName);
+
+    table.refresh();
+
+    String newDistributionMode = table.properties().get(TableProperties.WRITE_DISTRIBUTION_MODE);
+    assertThat(newDistributionMode).as("Distribution mode must match").isEqualTo("hash");
```

Comment:
rather than checking against hash here we should get the distribution mode before calling "alter" and compare against that.